sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -241,13 +241,29 @@ public async Task<IEnumerable<Lease>> GetAllLeasesAsync()
 
             do
             {
-                var leaseBlobsResult = await this.consumerGroupDirectory.ListBlobsSegmentedAsync(
+                var listBlobsTask = this.consumerGroupDirectory.ListBlobsSegmentedAsync(
                     true,
                     BlobListingDetails.Metadata,
                     null,
                     continuationToken,
                     this.defaultRequestOptions,
-                    this.operationContext).ConfigureAwait(false);
+                    this.operationContext);
+
+                // ListBlobsSegmentedAsync honors neither timeout settings in request options nor cancellation token and thus intermittently hangs.
+                // This provides a workaround until we have storage.blob library fixed.
+                await Task.WhenAny(listBlobsTask, Task.Delay(this.LeaseRenewInterval)).ConfigureAwait(false);
```

Comment:
This pattern causes a timer leak because the task returned by Task.Delay is never dispose. See 
- https://github.com/Azure/azure-sdk-for-net/pull/10019/files

Reply:
I'm not sure that link was what you were looking for.   I think you were trying to call attention to the [Azure.Core Task Extensions](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/TestFramework/TaskExtensions.cs#L54) that have an example of this pattern.

Reply:
My bad https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#using-a-timeout is the link

Reply:
It isn't a leak, just the pattern may cause large number of timers if the operation is so frequently called. In our case this is called only once per interval, so by default once every 10 seconds. Wouldn't event cause any issues.

Reply:
Btw, please note that this is temporary. Multiple customers are hitting this issue. Once Storage fixes it, we will revert this back.

Reply:
I will take in the recommended way of cancelling timer and send new iteration. Better to be on the safe side. Thanks @pakrym for pointing this.